cmd/posix.go

```diff
@@ -1547,8 +1552,11 @@ func (s *posix) RenameFile(srcVolume, srcPath, dstVolume, dstPath string) (err e
 
 func (s *posix) VerifyFile(volume, path string, fileSize int64, algo BitrotAlgorithm, sum []byte, shardSize int64) (err error) {
 	defer func() {
-		if err == errFaultyDisk {
-			atomic.AddInt32(&s.ioErrCount, 1)
+		if err != nil {
+			if err == errFaultyDisk {
+				atomic.AddInt32(&s.ioErrCount, 1)
+			}
+			err = VerifyFileError(err.Error())
```

Comment:
Healing will be more predictable (and safer) if it heals only on specific known errors - bitrot, filenotfound, corrupt

Hence it is better to encapsulate only these 3 errors in VerifyFileError.

Ex. for errVolumeNotFound healing code should not try healing.

Reply:
Fixed